subsys/usb/usb_device.c

```diff
@@ -863,14 +941,21 @@ static bool usb_handle_std_endpoint_req(struct usb_setup_packet *setup,
 		break;
 
 	case REQ_CLEAR_FEATURE:
-		if (setup->wValue == FEA_ENDPOINT_HALT) {
-			/* clear HALT by unstalling */
-			LOG_INF("... EP clear halt %x", ep);
-			usb_dc_ep_clear_stall(ep);
-			if (usb_dev.status_callback) {
-				usb_dev.status_callback(USB_DC_CLEAR_HALT, &ep);
+		if (sys_le16_to_cpu(setup->wValue) == FEA_ENDPOINT_HALT) {
+			/** This request is valid for Control Endpoints when
+			 * the device is not yet configured. For other
+			 * Endpoints the device must be configured.
+			 */
+			if (!(ep & ~USB_EP_DIR_MASK) ||
```

Comment:
`if (USB_EP_GET_IDX(ep) == 0 || ...` is much easier to read.

Reply:
Yea, I forgot we have those macros. I will update this. Generally I made this fix for `REQ_CLEAR_FEATURE` but now I can see similar check is required for other Endpoint specific requests.